clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -2209,12 +2193,7 @@ mlir::Value CIRGenFunction::buildLoadOfScalar(LValue lvalue,
                            lvalue.isNontemporal());
 }
 
-mlir::Value CIRGenFunction::buildFromMemory(mlir::Value Value, QualType Ty) {
-  // Bool has a different representation in memory than in registers.
-  if (hasBooleanRepresentation(Ty)) {
```

Comment:
Change this to `if (!Ty->isBooleanType() && hasBooleanRepresentation(Ty))` instead.